container-scan/action.yaml

```diff
@@ -1,152 +1,78 @@
+    description: "Lacework access token"
+  dockerhub-user:
     required: true
-    description: Docker image tag
-  image-platform:
-    description: Target platform to build image for (eg. linux/amd64 (default), linux/arm64, etc)
-    required: false
-    default: linux/amd64
-
-  build-args:
-    required: false
-    description: List of build arguments for docker build as key-value pairs (e.g., KEY=VALUE)
-    default: ""
-  secrets:
-    required: false
-    description: List of secrets for docker build as key-value pairs (e.g., SECRET_KEY=VALUE)
-    default: ""
-  enable-docker-build:
-    required: false
-    default: true
-    description: Docker image tag
+    description: "DockerHub username"
+  dockerhub-password:
+    required: true
+    description: "DockerHub password"
 
 outputs:
-  comment-id:
-    description: Comment ID of the test report
-    value: ${{ steps.comment-pr.outputs.comment-id }}
+  scan-results-path:
+    description: "Path to the scan results file"
+    value: ${{ steps.scan.outputs.results-file }}
 
 runs:
   using: composite
   steps:
     - name: Checkout Repository
       uses: actions/checkout@v4
 
-    - name: Build docker image
-      if: ${{ inputs.enable-docker-build }}
-      uses: open-turo/actions-security/docker-build@v3
+    - name: Build Docker Image
+      if: ${{ inputs.enable-docker-build == 'true' }}
+      uses: open-turo/actions-security/docker-build@v2
       id: docker-build
       with:
         dockerhub-user: ${{ inputs.dockerhub-user }}
         dockerhub-password: ${{ inputs.dockerhub-password }}
-        github-token: ${{ inputs.github-token }}
         image-version: ${{ inputs.image-tag }}
-        image-platform: ${{ inputs.image-platform }}
-        load: true
-        push: false
-        docker-metadata-tags: |
-          type=ref,event=branch
-          type=ref,event=pr
         build-args: ${{ inputs.build-args }}
-        secrets: ${{ inputs.secrets }}
 
-    - name: "Determining image name"
-      shell: bash
-      id: set_image_name
+    - name: Determine Image Name
       run: |
-        if [ "${{ inputs.enable-docker-build }}" == "true" ]; then
-          echo "::set-output name=image_name::${{ steps.docker-build.outputs.image-name }}"
+        if [ "${{ inputs.enable-docker-build }}" = "true" ]; then
+          echo "IMAGE_NAME=${{ steps.docker-build.outputs.image-name }}" >> $GITHUB_ENV
         else
-          echo "::set-output name=image_name::${{ inputs.image-name }}"
+          echo "IMAGE_NAME=${{ inputs.image-name }}" >> $GITHUB_ENV
         fi
+      shell: bash
 
-    - name: Scan container image for vulnerabilities using Lacework
+    - name: Scan Container Image
+      id: scan
       uses: lacework/lw-scanner-action@v1.4.3
       with:
         LW_ACCOUNT_NAME: ${{ inputs.lw-account-name }}
         LW_ACCESS_TOKEN: ${{ inputs.lw-access-token }}
-        IMAGE_NAME: ${{ steps.set_image_name.outputs.image_name }}
+        IMAGE_NAME: ${{ env.IMAGE_NAME }}
```

Comment:
Why turn this into an env variable? It seems using the step output was cleaner. Was there a specific reason?